ivy/functional/frontends/torch/nn/functional/pooling_functions.py

```diff
@@ -191,3 +191,30 @@ def lp_pool1d(input, norm_type, kernel_size, stride=None, ceil_mode=False):
         )
 
     return ivy.pow(ivy.multiply(out, kernel_size), ivy.divide(1.0, norm_type))
+
+
+@to_ivy_arrays_and_back
+def lp_pool2d(input, norm_type, kernel_size, stride=None, ceil_mode=False):
+
+    data_format = "NCHW"
+    padding = "VALID"
+    if stride is not None:
+        out = ivy.avg_pool2d(
+            ivy.pow(input, norm_type),
+            kernel_size,
+            stride,
+            padding,
+            data_format=data_format,
+        )
+    else:
+        out = ivy.avg_pool2d(
+            ivy.pow(input, norm_type),
+            kernel_size,
+            kernel_size,
+            padding,
+            data_format=data_format,
+        )
+
+    return ivy.pow(
+        ivy.multiply(out, kernel_size[0] * kernel_size[1]), ivy.divide(1.0, norm_type)
```

Comment:
We are assuming that `kernel_size` is a `tuple` here, this is not necessarily that case as it can be of type `int`.